### PR TITLE
Modernize notebook tests

### LINKED ISSUE #5173: Switch to Jest for Testing
I think we should switch to Jest for testing, including the `services` package.  JSDom supports both `WebSocket` and `requestAnimationFrame`.  We would avoid the extra layers of WebPack and Karma, running directly in Node.  I looked at our uses of `simulate-event`, and none of them are relying on a specific client position.

cc @saulshanabrook 

Remaining test suites to convert
- [x] console
- [x] csvviewer
- [x] docmanager
- [x] filebrowser
- [x] notebook
- [x] rendermime
- [x] terminal

### PR DESCRIPTION
**Pull request by blink1073**
## References
The *LAST PACKAGE* for #8124!!  Fixes #5173.

[Alt text](https://www.youtube.com/watch?v=9jK-NcRmVcw)

## Code changes
- Upgraded the test suite of `notebook`.  Changed to use the `JupyterServer` class for the toolbar and session context tests.  Adds some local shimming for `NBTestUtils` since `testutils` imports from `notebook/lib` not `notebook/src`.

## User-facing changes
Easier to contribute to this package.

## Backwards-incompatible changes
None

### CODE REVIEW

File: packages/notebook/test/tracker.spec.ts [hunk 1]
```diff
@@ -1,22 +1,31 @@
 // Copyright (c) Jupyter Development Team.
-// Distributed under the terms of the Modified BSD License.
```
Comment (afshin) [inline packages/notebook/test/tracker.spec.ts:2]: This probably was an accidental deletion.

File: packages/notebook/test/actions.spec.ts [hunk 2]
```diff
@@ -1,42 +1,59 @@
 // Copyright (c) Jupyter Development Team.
-// Distributed under the terms of the Modified BSD License.
```
Comment (afshin) [inline packages/notebook/test/actions.spec.ts:2]: This was likely an accidental deletion.

File: packages/notebook/test/model.spec.ts [hunk 3]
```diff
@@ -1,40 +1,41 @@
 // Copyright (c) Jupyter Development Team.
-// Distributed under the terms of the Modified BSD License.
```
Comment (afshin) [inline packages/notebook/test/model.spec.ts:2]: Ditto

File: packages/notebook/test/notebooktools.spec.ts [hunk 4]
```diff
@@ -1,7 +1,6 @@
 // Copyright (c) Jupyter Development Team.
-// Distributed under the terms of the Modified BSD License.
```
Comment (afshin) [inline packages/notebook/test/notebooktools.spec.ts:2]: Ditto

File: packages/notebook/test/widget.spec.ts [hunk 5]
```diff
@@ -1,7 +1,6 @@
 // Copyright (c) Jupyter Development Team.
-// Distributed under the terms of the Modified BSD License.
```
Comment (afshin) [inline packages/notebook/test/widget.spec.ts:2]: This

File: packages/notebook/test/widgetfactory.spec.ts [hunk 6]
```diff
@@ -1,24 +1,32 @@
 // Copyright (c) Jupyter Development Team.
-// Distributed under the terms of the Modified BSD License.
```
Comment (afshin) [inline packages/notebook/test/widgetfactory.spec.ts:2]: 👀

File: testutils/src/jest-shim.ts [hunk 7]
```diff
@@ -52,3 +52,12 @@ process.on('unhandledRejection', (error, promise) => {
   }
   promise.catch(err => console.error('promise rejected', err));
 });
+
+(window as any).getSelection = function getSelection() {
```
Comment (afshin) [inline testutils/src/jest-shim.ts:56]: Should this check if `window.getSelection()` already exists or is it safe to do what you've done here?

File: packages/notebook/test/tracker.spec.ts (hunk 1 above)
Comment (blink1073) [reply packages/notebook/test/tracker.spec.ts:2]: I've added this to https://github.com/jupyterlab/jupyterlab/issues/8327 as a follow-on task for all test files.

File: testutils/src/jest-shim.ts (hunk 7 above)
Comment (blink1073) [reply testutils/src/jest-shim.ts:56]: Given the limited scope we're using it for, I think this is fine.  Interestingly, newer versions of jsdom do have this method though.

**Pull request merged**